packages/flutter_tools/lib/src/hot.dart

```diff
@@ -325,13 +335,17 @@ class HotRunner extends ResidentRunner {
   Future<Null> handleTerminalCommand(String code) async {
     final String lower = code.toLowerCase();
     if ((lower == 'r') || (code == AnsiTerminal.KEY_F5)) {
+      OperationResult result = OperationResult.ok;
       // F5, restart
       if ((code == 'r') || (code == AnsiTerminal.KEY_F5)) {
         // lower-case 'r'
-        await _reloadSources();
+        result = await _reloadSources();
       } else {
         // upper-case 'R'.
-        await _restartFromSources();
+        result = await _restartFromSources();
+      }
+      if (!result.isOk) {
+        printStatus('Try again after fixing the above error(s).', emphasis: true);
```

Comment:
Bonus points if you can figure out how many errors there were, and use error/errors appropriately instead of "error(s)". :-)